internal/workspace/template_resolver.go

```diff
@@ -44,7 +44,7 @@ func (tr *TemplateResolver) ResolveTemplate(ctx context.Context, templateRef *wo
 	err := tr.client.Get(ctx, templateKey, template)
 
 	// If not found and we have a default namespace, try there
-	if err != nil && tr.defaultTemplateNamespace != "" && templateNamespace != tr.defaultTemplateNamespace {
+	if client.IgnoreNotFound(err) == nil && tr.defaultTemplateNamespace != "" && templateNamespace != tr.defaultTemplateNamespace {
```

Comment:
sanity-check: shouldn't you keep an `err != nil &&` first? as in, what would `client.IgnoreNotFound(err)` return if `err == nil` ?

For reference, we're using `errors.isNotFound()` elsewhere in the codebase ([example](https://github.com/jupyter-infra/jupyter-k8s/blob/8de30deb51b2c11564a3bd8d4574ea232a1c1e75/internal/controller/resource_manager.go#L191)), however no objection if this functionally works.